state/factory/statedb.go

```diff
@@ -332,3 +330,23 @@ func (sdb *stateDB) createGenesisStates(ctx context.Context) error {
 
 	return sdb.commit(ws)
 }
+
+func (sdb *stateDB) readFromWorkingSets(key hash.Hash256) (WorkingSet, error) {
+	sdb.mutex.RLock()
+	defer sdb.mutex.RUnlock()
+	if data, ok := sdb.workingsets.Get(key); ok {
```

Comment:
so the root-cause is need to use mutex for get/putWorkingSets?

Reply:
The root cause is that in runActions(), we shouldn't lock the sf mutex. But while detaching workingset, get/put workingset needs mutex to be locked so I changed to lock/unlock the mutex seperately if needed